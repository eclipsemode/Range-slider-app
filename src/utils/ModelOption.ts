declare type ModelOption = {
    from: number,
    to: number,
    gap: number,
    step: number,
    min: number,
    max: number,
    vertical: boolean;
    rulers: boolean,
    range: boolean,
    tooltip: boolean,
    percent: boolean,
    color: {
        firstColor: string,
        secondColor: string,
        textColor: string,
        thumbColor: string
    },
    configPanel: boolean,
    controlConfig: string[],
    toggleConfig: string[],
    progress: boolean
}

export default ModelOption;