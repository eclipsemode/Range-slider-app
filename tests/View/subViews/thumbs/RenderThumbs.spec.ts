import $ from "jquery";
import {
  CreateBar,
  CreateProgress,
  CreateThumbFrom,
  CreateThumbTo,
  RenderThumbs,
} from "../../../../src/View/subViews";
import ModelOption from "../../../../src/utils/ModelOption";

describe("Render Thumbs tests", () => {
  const body: JQuery = $("body");
  const bar: CreateBar = new CreateBar(body);
  const thumbFrom: CreateThumbFrom = new CreateThumbFrom(body);
  const thumbTo: CreateThumbTo = new CreateThumbTo(body);
  const options: ModelOption = {
    max: 104,
    tooltip: true,
    range: true,
    configPanel: true,
    step: 33,
    gap: 1,
    min: 0,
    from: 0,
    to: 99,
    vertical: false,
    rulers: true,
    percent: false,
    color: {
      firstColor: "#ffe53b",
      secondColor: "#ff2525",
      textColor: "#ffe53b",
      thumbColor: "#ffe53b",
    },
    progress: true,
    controlConfig: ["min", "max", "step", "from", "to"],
    toggleConfig: ["vertical", "range", "rulers", "progress", "tooltip"],
  };
  const progress: CreateProgress = new CreateProgress(body);
  test("Should be defined", () => {
    const renderThumbs: RenderThumbs = new RenderThumbs(
      options,
      thumbFrom,
      thumbTo,
      bar,
      progress
    );
    expect(renderThumbs.fromThumb).toBeDefined();
    expect(renderThumbs.toThumb).toBeDefined();
  });
  test("Should match vertical", () => {
    const renderThumbs: RenderThumbs = new RenderThumbs(
      { ...options, vertical: true },
      thumbFrom,
      thumbTo,
      bar,
      progress
    );
    expect(`${renderThumbs.fromThumb.fromThumbElement[0].classList}`).toMatch(
      "vertical"
    );
  });
  test("Should match snapshot", () => {
    const renderThumbs: RenderThumbs = new RenderThumbs(
      options,
      thumbFrom,
      thumbTo,
      bar,
      progress
    );
    expect(renderThumbs.toThumb.toThumbElement[0].innerHTML).toMatchSnapshot(
      'class="slider-app__thumb'
    );
  });
  test("Should be a null to thumb", () => {
    const renderThumbs: RenderThumbs = new RenderThumbs(
      { ...options, range: false },
      thumbFrom,
      thumbTo,
      bar,
      progress
    );
    expect(renderThumbs.toThumb).toBeNull();
  });
});
