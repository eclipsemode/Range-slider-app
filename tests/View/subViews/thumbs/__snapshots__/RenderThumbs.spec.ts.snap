// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Thumbs tests Should match snapshot: class="slider-app__thumb 1`] = `""`;

exports[`Render Thumbs tests Should match snapshot: ss 1`] = `
jQuery {
  "0": <div
    class="slider-app__thumb slider-app__thumb-to"
    style="left: 0px;"
  />,
  "length": 1,
}
`;

exports[`Render Thumbs tests Should match snapshot: ssaddasdafcas 1`] = `
jQuery {
  "0": <div
    class="slider-app__thumb slider-app__thumb-to"
    style="left: 0px;"
  />,
  "length": 1,
}
`;
